src/github_changelog/github.clj

```diff
@@ -14,39 +14,45 @@
   (let [{:keys [github-api user repo]} config]
     (format "%s/repos/%s/%s/pulls" (strip-trailing github-api "/") user repo)))
 
-(defn- make-request [config params]
-  (let [oauth-token (:token config)]
-    {:as           :json
-     :query-params (merge {:state "closed"} params)
-     :headers      {"User-Agent"    "GitHub-Changelog"
-                    "Authorization" (str "token " oauth-token)}}))
-
-(def call-api (throttle-fn http/get 5 :second))
+(defn- make-request
+  ([config] (make-request config {}))
+  ([config params]
+   (let [oauth-token (:token config)]
+     {:as           :json
+      :query-params (merge {:state "closed"} params)
+      :headers      {"User-Agent"    "GitHub-Changelog"
+                     "Authorization" (str "token " oauth-token)}})))
 
 (defn- last-page-number [links]
-  (-> (get-in links [:last :href])
-      (split #"\?")
-      second
-      extract-params
-      :page
-      Integer/parseInt))
+  (some-> (get-in links [:last :href])
+          (split #"\?")
+          second
+          extract-params
+          :page
+          Long/parseLong))
 
 (defn- gen-pages [links]
-  (range 2 (inc (last-page-number links))))
-
-(defn- get-pulls [config]
-  (let [end-point (pulls-url config)
-        request (make-request config {})
-        first-response (call-api end-point request)
-        links (:links first-response)
-        first-body (:body first-response)]
-    (if links
-      (let [pages (gen-pages links)
-            requests (map #(make-request config {:page %}) pages)
-            rest-responses (pmap #(call-api end-point %) requests)]
-        (into first-body (flatten (map :body rest-responses))))
-      first-body)))
+  (if-let [last-page (last-page-number links)]
+    (range 2 (inc last-page))
+    []))
```

Comment:
I read a bit about it and it is totally fine to return nil, where you want to supply an empty sequence.
(Since Clojure handles `nil` so well)
With that:
```clojure
(when-let [last-page (last-page-number links)]
  (range 2 (inc last-page)))
```